File: 215-kth-largest-element-in-an-array/kth-largest-element-in-an-array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findKthLargest = function(nums, k) {
    const maxHeap = new MaxPriorityQueue()

    for (const num of nums) {
        maxHeap.enqueue(num)
    }

    for (let i = 1; i < k; i++) {
        maxHeap.dequeue()
    }

    return maxHeap.dequeue().element
};

File: 215-kth-largest-element-in-an-array/kth-largest-element-in-an-array.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        heap = []

        for num in nums:
            heapq.heappush(heap, -num)

        for _ in range(k - 1):
            heapq.heappop(heap)

        return -heap[0]